beetsplug/parentwork.py

```diff
@@ -0,0 +1,197 @@
+        """Import hook for fetching parent works automatically.
+        """
+        force_parent = self.config['force'].get(bool)
+
+        for item in task.imported_items():
+            self.find_work(item, force_parent)
+            item.store()
+
+    def get_info(self, item, work_info):
+        """Given the parentwork info dict, this function fetches
+        parent_composer, parent_composer_sort, parentwork,
+        parentwork_disambig, mb_workid and composer_ids"""
+
+        parent_composer = []
+        parent_composer_sort = []
+
+        composer_exists = False
+        if 'artist-relation-list' in work_info['work']:
+            for artist in work_info['work']['artist-relation-list']:
+                if artist['type'] == 'composer':
+                    parent_composer.append(artist['artist']['name'])
+                    parent_composer_sort.append(artist['artist']['sort-name'])
+        if not composer_exists:
+            self._log.info(item.artist + ' - ' + item.title)
+            self._log.debug(
+                "no composer, add one at https://musicbrainz.org/work/" +
+                work_info['work']['id'])
+        parentwork = work_info['work']['title']
+        mb_parentworkid = work_info['work']['id']
+        if 'disambiguation' in work_info['work']:
+            parentwork_disambig = work_info['work']['disambiguation']
+            return [parentwork, mb_parentworkid, parent_composer,
+                    parent_composer_sort, parentwork_disambig]
+        else:
+            return [parentwork, mb_parentworkid, parent_composer,
+                    parent_composer_sort, None]
+
+    def find_work(self, item, force):
+
+        recording_id = item.mb_trackid
+        try:
+            item.parentwork
+            hasparent = True
+        except AttributeError:
+            hasparent = False
+        if not item.mb_workid:
+            self._log.info("No work attached, recording id: " +
+                           recording_id)
+            self._log.info(item.artist + ' - ' + item.title)
+            self._log.info("add one at https://musicbrainz.org" +
+                           "/recording/" + recording_id)
+            return
+        found = False
+        if force or (not hasparent):
+            try:
+                work_info, work_date = find_parentwork(item.mb_workid)
+                parent_info = self.get_info(item, work_info)
+                parentwork = parent_info[0]
+                mb_parentworkid = parent_info[1]
+                parent_composer = parent_info[2]
+                parent_composer_sort = parent_info[3]
+                parentwork_disambig = parent_info[4]
+
+                found = True
+            except musicbrainzngs.musicbrainz.WebServiceError:
+                self._log.debug("Work unreachable")
+                found = False
+        elif hasparent:
+            self._log.debug("Work already in library, not necessary fetching")
+            return
+
+        if found:
```

Comment:
I don't think you need `found` or this conditional—just put this right in the condition above after gathering the data. (Or perhaps more appropriately, after an "early return.")